api/internal/handler/global_rule/global_rule_test.go

```diff
@@ -0,0 +1,338 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package global_rule
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/shiningrush/droplet"
+	"github.com/shiningrush/droplet/data"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/apisix/manager-api/internal/core/entity"
+	"github.com/apisix/manager-api/internal/core/store"
+)
+
+func performRequest(r http.Handler, method, path string) *httptest.ResponseRecorder {
+	req := httptest.NewRequest(method, path, nil)
+	w := httptest.NewRecorder()
+	r.ServeHTTP(w, req)
+	return w
+}
+
+func TestHandler_ApplyRoute(t *testing.T) {
+	mStore := &store.MockInterface{}
+	giveRet := `{
+		"id": "test",
+		"plugins": {
+			"jwt-auth": {}
+		}
+	}`
+	mStore.On("Get", mock.Anything).Run(func(args mock.Arguments) {
+		assert.Equal(t, "test", args.Get(0))
+	}).Return(giveRet, nil)
+
+	h := Handler{globalRuleStore: mStore}
+	r := gin.New()
+	h.ApplyRoute(r)
+
+	w := performRequest(r, "GET", "/apisix/admin/global_rules/test")
+	assert.Equal(t, 200, w.Code)
```

Comment:
I remember we will response 200 with code if an error occurred, it seems that the Get method relies on etcd go client?

Reply:
It just happened when binding params to struct error.
and it's fixed by #899